slash/core/fixtures/active_fixture.py

```diff
@@ -33,10 +36,19 @@ def call_test_end(self):
             for callback in self._test_end_callbacks:
                 callback()
 
-    def add_cleanup(self, cleanup):
-        self._cleanups.append(cleanup)
+    def add_cleanup(self, cleanup=None, success_only_cleanup=False):
+        if cleanup is None:
+            return functools.partial(self.add_cleanup, success_only_cleanup=success_only_cleanup)
+
+        self._cleanups.append((cleanup, success_only_cleanup))
 
     def do_cleanups(self):
+        if not slash_context.result:
+            # for when we're unit-testing, and there's not 'result'
+            success = True
+        else:
+            success = slash_context.result.is_success()
```

Comment:
@vmalloc - maybe there's a better way of determining `success`? it wasn't compatible with `test_fixture_scoping.py`